.github/workflows/dependency-review.yml

```diff
@@ -0,0 +1,28 @@
+---
+name: Dependency Review
```

Comment:
We have a convention regarding our workflow names. It isn't well-published, so I can understand why this was missed. 

https://github.com/ministryofjustice/operations-engineering?tab=readme-ov-file#naming-standards-for-workflow-files